File: src/service/ModelBuilder.Models/Application/ProjectAm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Mb.Models.Extensions;

namespace Mb.Models.Application;

public class ProjectAm : IValidatableObject
{
    
    public Guid Id { get; set; }

    [Required]
    public string Version { get; set; }

    [Required]
    public string Name { get; set; }

    public string Description { get; set; }

    [Required]
    public bool SubProject { get; set; }

    public string CreatedBy { get; set; }
    public DateTime Created { get; set; }
    public string UpdatedBy { get; set; }
    public DateTime? Updated { get; set; }

    public ICollection<BlockAm> Blocks { get; set; }
    public ICollection<ConnectionAm> Connections { get; set; }

    #region Validate

    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
    {
        var validations = new List<ValidationResult>();
        validations.AddRange(Blocks.Validate(validationContext));
        validations.AddRange(Connections.Validate(validationContext));
        return validations;
    }

    #endregion Validate

    #region Public Methods

    public IEnumerable<ConnectionAm> GetParentlessConnectors()
    {
        return Connections.GetParentlessConnectors(Blocks);
    }

    public IEnumerable<ConnectionAm> GetNotConnectedConnectors()
    {
        return Connections.GetNotConnectedConnectors(Blocks);
    }

    #endregion
}

File: src/client/src/lib/classes/application/ProjectAm.ts
import { Project, AspectObjectAm, ConnectionAm } from "..";

export class ProjectAm {
  id: string;
  name: string;
  isSubProject: boolean;
  version: string;
  description: string | null;
  updatedBy: string | null;
  updated: Date | null;
  created: Date;
  createdBy: string;
  aspectObjects: AspectObjectAm[] | null;
  connections: ConnectionAm[] | null;

  public constructor(obj: Project) {
    this.id = obj.id;
    this.name = obj.name;
    this.isSubProject = obj.isSubProject;
    this.version = obj.version;
    this.description = obj.description;
    this.updatedBy = obj.updatedBy;
    this.updated = obj.updated;
    this.created = obj.created;
    this.createdBy = obj.createdBy;
    this.aspectObjects = obj.aspectObjects.map((x) => new AspectObjectAm(x));
    this.connections = obj.connections.map((x) => new ConnectionAm(x));
  }
}
